contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,817 @@
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    uint256 domainSkillId = colony.getDomain(_domainId).skillId;
+    uint256 actionDomainSkillId = getActionDomainSkillId(_action);
+
+    if (domainSkillId != actionDomainSkillId) {
+      uint256 childSkillId = colonyNetwork.getChildSkillId(domainSkillId, _childSkillIndex);
+      require(childSkillId == actionDomainSkillId, "voting-rep-invalid-domain-id");
+    }
+
+    createPoll(_target, _action, _domainId, domainSkillId, _key, _value, _branchMask, _siblings);
+  }
+
+  function stakePoll(
+    uint256 _pollId,
+    uint256 _permissionDomainId, // For extension's arbitration permission
+    uint256 _childSkillIndex, // For extension's arbitration permission
+    uint256 _vote,
+    uint256 _amount,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(_vote <= 1, "voting-rep-bad-vote");
+    require(getPollState(_pollId) == PollState.Staking, "voting-rep-staking-closed");
+
+    uint256 requiredStake = getRequiredStake(_pollId);
+
+    // Either we are before the commit deadline or other side is fully staked
+    require(
+      now < add(poll.events[CREATE], wmul(stakePeriod, STAKE_COMMIT_FRACTION)) ||
+      poll.stakes[flip(_vote)] == requiredStake,
+      "voting-rep-cannot-stake"
+    );
+
+    uint256 amount = min(_amount, sub(requiredStake, poll.stakes[_vote]));
+    uint256 stakerTotalAmount = add(stakes[_pollId][msg.sender][_vote], amount);
+
+    require(
+      stakerTotalAmount <= checkReputation(_pollId, msg.sender, _key, _value, _branchMask, _siblings),
+      "voting-rep-insufficient-rep"
+    );
+    require(
+      stakerTotalAmount >= wmul(requiredStake, minStakeFraction),
+      "voting-rep-insufficient-stake"
+    );
+
+    colony.obligateStake(msg.sender, poll.domainId, amount);
+    colony.transferStake(_permissionDomainId, _childSkillIndex, address(this), msg.sender, poll.domainId, amount, address(this));
+
+    // Update the stake
+    poll.unpaidRewards = add(poll.unpaidRewards, amount);
+    poll.stakes[_vote] = add(poll.stakes[_vote], amount);
+    stakes[_pollId][msg.sender][_vote] = stakerTotalAmount;
+
+    // Increment counter & extend claim delay if staking for an expenditure state change
+    if (poll.stakes[YAY] == requiredStake && _vote == YAY && getSig(poll.action) == CHANGE_FUNCTION) {
+      bytes32 structHash = hashExpenditureStruct(poll.action);
+      expenditurePollCounts[structHash] = add(expenditurePollCounts[structHash], 1);
+      bytes memory claimDelayAction = createClaimDelayAction(poll.action, UINT256_MAX);
+      executeCall(_pollId, claimDelayAction);
+    }
+
+    // Claim tokens if fully staked
+    if (poll.stakes[YAY] == requiredStake && poll.stakes[NAY] == requiredStake) {
+      poll.events[STAKE_END] = now;
+      tokenLocking.claim(token, true);
```

Comment:
Really? I would have thought that a reward claiming tx would have failed... I'll look into it.

Reply:
`claim` is only needed on token locking when a token is locked, which currently only happens when a payout in a colony is started. When the token is locked and tokens are transferred, it goes in to `pendingBalance`, and `claim` moves from there in to `balance`. Otherwise, transferred tokens go straight in to `balance`. As I don't think any tests start a colony reward payout mid-vote, this doesn't come up.